src/main/scala/org/renci/cam/QueryService.scala

```diff
@@ -34,16 +53,15 @@ object QueryService extends LazyLogging {
 
   private def convertCase(v: String): String = CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, v)
 
-  def getNodeTypes(nodes: List[TRAPIQueryNode]): Map[String, String] = {
+  def getNodeTypes(nodes: List[TRAPIQueryNode]): Map[String, IRI] =
     nodes
       .map(node =>
         (node.`type`, node.curie) match {
-          case (_, Some(c))    => (node.id, c.value)
-          case (Some(t), None) => (node.id, t.iri.value)
-          case (None, None)    => (node.id, "")
+          case (_, Some(c))    => (node.id, c)
+          case (Some(t), None) => (node.id, t.iri)
+          case (None, None)    => (node.id, IRI(""))
```

Comment:
This `IRI("")` seems like a problem. Does the return type need to be `Option[IRI]` instead of `IRI`?

Reply:
If I am not mistaken, a TRAPIQueryNode can't have both a None type & a None curie.  So I am thinking that (None, None) should throw an exception.  Would that work?